docs/design/2023-11-29-priority-queue-for-auto-analyze.md

```diff
@@ -0,0 +1,270 @@
+
+   1. We still use tidb_auto_analyze_ratio to determine if the table needs to be analyzed. The default value is 0.5. It means that auto-analyze is triggered when greater than 50% of the rows in a table have been modified.
+   2. For special events, it is necessary to put the table into the priority queue. For example, the table has a new index but it hasn't been analyzed yet.
+   3. If a table is added to the auto-analysis queue, a weight must be assigned to it to determine the execution order.
+
+Weight table:
+| Name                 | Meaning                                                                 | Weight                                                                                                                                                                                                                                                                                                                                                            |
+|----------------------|-------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Percentage of Change | The percentage of change since the last analysis.                       | 50% <= x < 70%: 1 <br/> 70% <= x < 80%: 1.3 <br/> x >= 80%: 1.5  <br/>                                                                                                                                                                                                                                                                                            |
+| Last Failed Interval | The interval between now and the last failed analysis ends time.        | Here we cannot easily determine what interval is long and what interval is short. Therefore, we can decide by querying the interval between each automatic analysis in history. <br/> interval < average automatic analysis interval: 0 <br/> interval >=  average automatic analysis interval: 0.5 <br/> interval >=  2 * average automatic analysis interval: 1 |
+| Special Event        | For example, the table has a new index but it hasn't been analyzed yet. | HasNewIndexWithoutStats: 1.5 <br/> TableNeverBeenAnalyzed: 1.5                                                                                                                                                                                                                                                                                                    |
+
+The calculation formula is: `priority_score = (change_percentage[size] * last_failed_time_weight[interval] * special_event[event])`
```

Comment:
So we calculate the score of each table by using this formula each time when triggering auto-analyze, and pick the table with the highest score to analyze?

Reply:
Just want to remind that don't forget to log this formula and scores each time when triggering auto-analyze.

Reply:
> So we calculate the score of each table by using this formula each time when triggering auto-analyze, and pick the table with the highest score to analyze?

Yes. 



> Just want to remind that don't forget to log this formula and scores each time when triggering auto-analyze.

I will add it.